### ISSUE TITLE
Use of finite source direct is *not* transparent

### ISSUE THREAD

**Issue by jenniferyee** [rpoleski/MulensModel#45]
In the finite source calculations, if the user selects "finite_source_uniform_Gould94", but the range it is supposed to be applied to extend beyond z=10, the code directly calculates the B0, B1 terms. This is very slow and not worthwhile because beyond z=10, finite source effects don't really matter.

Issue 1: the code does not warn the user when this happens. The calculation just slows down by a factor of 20, and the user is left to puzzle over why.

Issue 2: In the finite_source_gradient branch, I have extended the table to z = 9900. While this seems silly, it allows the fast interpolation for any value of the finite source, which bypasses Issue 1. However, if I change pointlens.py to use this file (v3 instead of v1), the unit test test_model_binary_and_finite_sources in test.Model.py fails. No other tests fail.

**Comment by rpoleski** [comment]
Issue 1 - we can just add a warning.

Issue 2 - I've added in that test:

```python
        mask = np.where(np.abs(fitted-expected) > 1e-7)[0]
        print(time[mask])
        print(fitted[mask])
        print(expected[mask])
        print(((fitted-expected)/fitted)[mask])
```
and got:
```python
[4999.95236961 5000.02381519 5099.97618481 5100.04763039]
[ 47.70564479  50.13291555 991.80582882 411.40351648]
[ 47.68486282  50.11215659 991.79675383 411.39443261]
[4.35629260e-04 4.14078343e-04 9.14997472e-06 2.20801840e-05]
```
So the differences are very small, but of course, we should track down their origin.

**Comment by jenniferyee** [comment]
These issues have been fixed as of [v2.18.0](https://github.com/rpoleski/MulensModel/releases/tag/v2.18.0), possibly earlier.
